instructor/common.py

```diff
@@ -97,66 +97,71 @@ def assignment_setup(request):
                                   'error': error,
                                   'assignment_name': assignment_name,
                                   'based_on': request.session['based_on'],
-                                  'new': request.session['new']
+                                  'new': request.session['new'],
                               },
                               context_instance=RequestContext(request))
 
 @login_required
 def course_setup(request):
-    CourseForm = modelform_factory(models.Course, fields=['code', 'name'])
 
-    if request.method == "POST":
-        if 'course_pk' in request.POST:
-            # Adding to an existing course
-            course = models.Course.objects.get(pk=request.POST['course_pk'])
-        else:
-            # Create a new course
+    CourseFormSet = modelformset_factory(models.Course, extra=1, can_delete=True, fields=['name', 'code'])
+    course_selected = 0
+    if request.method == 'POST':
+
+        # change this course's name or code
+        formset = CourseFormSet(request.POST)
+        if formset.is_valid():
+            courses = formset.save(commit=False)
+            for obj in formset.deleted_objects:
+                obj.delete()
```

Comment:
It looks like if you delete the last course in an assignment the assignment is deleted too, probably through [delete cascading](https://docs.djangoproject.com/en/1.8/ref/models/fields/#django.db.models.ForeignKey.on_delete). It may be simplest to just turn it off in the models but I'm not sure how that would affect everything else.